nilearn/decoding/tests/test_decoder.py

```diff
@@ -0,0 +1,311 @@
+    
+    for estimator in unsupported_estimators:
+        pytest.raises(ValueError, _check_estimator,
+                      _BaseDecoder(estimator=estimator).estimator)
+    custom_estimator = random_forest
+    pytest.warns(UserWarning, _check_estimator,
+                 _BaseDecoder(estimator=custom_estimator).estimator)
+
+
+def test_parallel_fit():
+    """The goal of this test is to check that results of _parallel_fit is the
+    same for different controlled param_grid
+    """
+
+    X, y = make_regression(n_samples=100, n_features=20,
+                           n_informative=5, noise=0.2, random_state=42)
+    train = range(80)
+    test = range(80, len(y_classification))
+    outputs = []
+    estimator = svr
+    svr_params = [[1e-1, 1e0, 1e1], [1e-1, 1e0, 5e0, 1e1]]
+    # define a scorer
+    scorer = check_scoring(estimator, 'r2')
+
+    for params in svr_params:
+        param_grid = {}
+        param_grid['C'] = np.array(params)
+        outputs.append(list(_parallel_fit(estimator=estimator, X=X, y=y,
+                                          train=train, test=test,
+                                          param_grid=param_grid,
+                                          is_classification=False, 
+                                          scorer=scorer,
+                                          mask_img=None, class_index=1,
+                                          screening_percentile=None)))
+    # check that every element of the output tuple is the same for both tries
+    for a, b in zip(outputs[0], outputs[1]):
+        if isinstance(a, np.ndarray):
+            np.testing.assert_array_almost_equal(a, b)
+        else:
+            assert a == b
+
+
+def test_decoder_binary_classification():
+    X, y = make_classification(n_samples=200, n_features=125, scale=3.0,
+                               n_informative=5, n_classes=2, random_state=42)
+    X, mask = to_niimgs(X, [5, 5, 5])
+
+    # check classification with masker object
+    model = Decoder(mask=NiftiMasker())
+    model.fit(X, y)
+    y_pred = model.predict(X)
+    assert accuracy_score(y, y_pred) > 0.95
+
+    # decoder object use predict_proba for scoring with logistic model
+    model = Decoder(estimator='logistic_l2', mask=mask)
+    model.fit(X, y)
+    y_pred = model.predict(X)
+    assert accuracy_score(y, y_pred) > 0.95
+
+    # check different screening_percentile value
+    for screening_percentile in [100, 20]:
+        model = Decoder(mask=mask, screening_percentile=screening_percentile)
+        model.fit(X, y)
+        y_pred = model.predict(X)
+        assert accuracy_score(y, y_pred) > 0.95
+
+    # check cross-validation scheme and fit attribute with groups enabled
+    for cv in [KFold(n_splits=5), LeaveOneGroupOut()]:
+        model = Decoder(estimator='svc', mask=mask,
+                        standardize=True, cv=cv)
+        if isinstance(cv, LeaveOneGroupOut):
+            groups = rand.binomial(2, 0.3, size=len(y))
```

Comment:
We should be using a local random state, and not the module-level one here, as it creates dependencies across tests.